network/gossip/libp2p/middleware.go

```diff
@@ -306,12 +306,6 @@ func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier)
 		return fmt.Errorf("failed to send message to %s: %w", targetID.String(), err)
 	}
 
-	// flush the stream
-	err = bufw.Flush()
```

Comment:
there is a flush at line 319. This flush cleans the buffer and the `bytecount` on line becomes 0 and consequently the outbound message size reported becomes 0.